docs/madr/decisions/055-default-hostname-generators.md

```diff
@@ -0,0 +1,218 @@
+We can remove the fallback if we add `kuma.io/display-name` to Universal resources by default.
+It's better than `{{ label "kuma.io/display-name" }}`, because looking at Kube resources it's annotation, not a label, so it might be confusing for users.
+
+### MeshService
+
+#### Local Kubernetes MeshService
+
+It's rather unlikely for user to use custom hostnames to consume MeshService that is local in the Kubernetes cluster.
+Therefore, no default HostnameGenerator is needed for it.
+
+#### Synced Kubernetes MeshService
+
+```yaml
+spec:
+  template: '{{ label "k8s.kuma.io/service-name" }}.{{ label "k8s.kuma.io/namespace" }}.svc.mesh.{{ label "kuma.io/zone" }}'
+  selector:
+    meshService:
+      matchLabels:
+        kuma.io/env: kubernetes # short term "kuma.io/managed-by: kube-controller"
+        kuma.io/origin: global # only select synced resources
+```
+
+It's important to only select Kubernetes MeshServices, so we won't try and fail to evaluate this template for Universal MeshServices.
+MeshService controller will add `kuma.io/managed-by: kube-controller` to all services created by it.
+It means that short term we won't take into account manually applied MeshServices on Kubernetes.
+This is not something we want to recommend to users anyway. They should use Service object. 
+
+This HostnameGenerator could be automatically created by Global CP so that it's synced down to every Zone CP.
+
+#### Local Universal MeshService
+
+```yaml
+spec:
+  template: '{{ .Name }}.svc.mesh.local' # .DisplayName would work as well
+  selector:
+    meshService:
+      matchLabels:
+        kuma.io/origin: zone # only take into account local MeshServices
+```
+
+Universal zones do not have hostnames, so we need to create HostnameGenerator for them.
+
+This HostnameGenerator could be automatically created by Universal Zone CP.
+
+#### Synced Universal MeshService
+
+```yaml
+spec:
+  template: '{{ .DisplayName }}.svc.mesh.{{ label "kuma.io/zone" }}'
+  selector:
+    meshService:
+      matchLabels:
+        kuma.io/origin: global
+        kuma.io/env: universal # short term the users need to apply it themselves 
+```
+
+This HostnameGenerator could be automatically created by Global CP so that it's synced down to every Zone CP.
+
+### MeshExternalService
+
+#### MeshExternalService applied on Zone CP
+
+```yaml
+spec:
+  template: '{{ .DisplayName }}.svc.meshext.local'
+  selector:
+    meshExternalService:
+      matchLabels:
+        kuma.io/origin: zone # only consider local MeshExternalServices
+```
+
+This HostnameGenerator could be automatically created by Zone CP.
+
+On Kubernetes, MeshExternalService can only be applied in the system namespace, so for example hostname for this MeshExternalService
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: httpbin
+  namespace: kuma-system
+```
+
+Would be `httpbin.svc.meshext.local`
+
+#### MeshExternalService applied on Global CP
+
+```yaml
+spec:
+  template: '{{ .DisplayName }}.svc.meshext.global'
```

Comment:
other alternative would be
```
{{ .DisplayName }}.extsvc.mesh.global
```
and
```
'{{ .DisplayName }}.mzsvc.mesh.global'
```
for meshmultizoneservice

Reply:
I like these two more 👍